promise.ml

```diff
@@ -1,58 +1,63 @@
 open Js_of_ocaml
 
-type +'a t = < > Js.t
+type +'a t = private < > Js.t
 
 type error = Js_of_ocaml.Js.Unsafe.any
 
+let js_value (promise : 'a t) : < > Js.t = (promise : 'a t :> < > Js.t)
+
 let promise_constr = Js.Unsafe.global##._Promise
 
 let indirect_promise_constr = Js.Unsafe.global##._IndirectPromise
 
 let wrap (value : 'a) : 'a = indirect_promise_constr##wrap value
 
-let unwrap (value : 'a) : 'a = indirect_promise_constr##unwrap value
+let unwrap (value : 'a) : 'b = indirect_promise_constr##unwrap value
 
 let make (f : resolve:('a -> unit) -> reject:('e -> unit) -> unit) : 'a t =
```

Comment:
Here, the type signature is also fragile. It should be written with locally abstract types or explicitly polymorphic type variables.

Reply:
I haven't used locally abstract types a lot.

How do you know when the type signature is considered fragile?

Reply:
In this type annotation `(x : 'a t)`, the `'a` can be unified with any type. That is, the compiler can freely choose a substitution for `'a` to make the typing rule hold.

When you use a locally abstract type variable, or an explicitly polymorphic annotation, the compiler will not have that freedom.

E.g.
```
utop # let x : 'a = 5;;
val x : int = 5

utop # let x : 'a. 'a = 5;;
Line 1, characters 17-18:
Error: This definition has type int which is less general than 'a. 'a

utop # let x (type a) : a = 5;;
Line 1, characters 21-22:
Error: This expression has type int but an expression was expected of type a
```